### PR TITLE
Updated S3.Event

### PR DESCRIPTION
**Pull request by mufumade**
### Motivation:

#194 

### Modifications:

Changed the S3.Event object size field to an optional due to the missing field on a s3:ObjectRemoved:* event
Added one additional test to check propper json decoding

### Result:

s3 event json will now decode successfully even if aws does not populate the `size` field.

### CODE REVIEW

File: Sources/AWSLambdaEvents/S3.swift [hunk 1]
```diff
@@ -69,7 +69,7 @@ public enum S3 {
 
     public struct Object: Codable {
         public let key: String
-        public let size: UInt64
+        public let size: UInt64? // A S3 ObjectRemoved:* event does not contain the object size
```
Comment (fabianfett) [inline Sources/AWSLambdaEvents/S3.swift:72]:
```suggestion
        /// The object's size in bytes. 
        ///
        /// Note: This property is available for all event types except "ObjectRemoved:*"
        public let size: UInt64?
```

File: Tests/AWSLambdaEventsTests/S3Tests.swift [hunk 2]
```diff
@@ -85,4 +126,32 @@ class S3Tests: XCTestCase {
...
+        XCTAssertEqual(record.eventTime, Date(timeIntervalSince1970: 1_578_907_540.621))
+        XCTAssertEqual(record.userIdentity, S3.UserIdentity(principalId: "AWS:AAAAAAAJ2MQ4YFQZ7AULJ"))
+        XCTAssertEqual(record.requestParameters, S3.RequestParameters(sourceIPAddress: "123.123.123.123"))
+        XCTAssertEqual(record.responseElements.count, 2)
+        XCTAssertEqual(record.s3.schemaVersion, "1.0")
+        XCTAssertEqual(record.s3.configurationId, "98b55bc4-3c0c-4007-b727-c6b77a259dde")
+        XCTAssertEqual(record.s3.bucket.name, "eventsources")
+        XCTAssertEqual(record.s3.bucket.ownerIdentity, S3.UserIdentity(principalId: "AAAAAAAAAAAAAA"))
+        XCTAssertEqual(record.s3.bucket.arn, "arn:aws:s3:::eventsources")
+        XCTAssertEqual(record.s3.object.key, "Hi.md")
```
Comment (fabianfett) [inline Tests/AWSLambdaEventsTests/S3Tests.swift:153]: Maybe we want to explicitly check that `object.size` is nil?
```suggestion
        XCTAssertEqual(record.s3.object.key, "Hi.md")
        XCTAssertNil(record.s3.object.size)
```

**Review by fabianfett — COMMENTED** [review]
Thanks for going ahead! Two small suggestions.

**Comment by mufumade** [comment]
@fabianfett Thanks for the feedback!

**Comment by fabianfett** [comment]
@swift-server-bot test this please

**Comment by fabianfett** [comment]
@mufumade Would you mind running `swiftformat .` once in the project folder, to please the soundness check?

**Comment by mufumade** [comment]
So this was a bit strange. At first swiftformat changed like 30 files although I only changed 2. On the second attempt swiftformat changed only the comment you provided with your suggestions. I hope it's all fine now.

**Comment by fabianfett** [comment]
@mufumade oh. sorry. it looks to me like our `swiftformat` is a little out of date. I'm updating it right now and will post another pr. Once that swiftformat pr has landed, we update this pr and run `swiftformat` again, which then should only affect your changed files. Sorry for the hassle and confusion.

**Comment by fabianfett** [comment]
For reference: https://github.com/swift-server/swift-aws-lambda-runtime/pull/196

**Comment by fabianfett** [comment]
@mufumade Would you mind resolving the merge conflicts and running swiftformat again? After that we should be ready to merge.

**Comment by tomerd** [comment]
@swift-server-bot test this please

**Comment by mufumade** [comment]
@fabianfett have a look at the newest commit.

**Comment by fabianfett** [comment]
@swift-server-bot test this please

File: Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift [hunk 3]
```diff
@@ -22,7 +22,7 @@ import NIOHTTP1
 /// * /runtime/invocation/error
 /// * /runtime/init/error
 extension Lambda {
-    internal struct RuntimeClient {
+    struct RuntimeClient {
```
Comment (fabianfett) [inline Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift:25]: @tomerd I have no idea, why `swiftformat` now removes the `internal` markers... Do we want to explicitly disable https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#redundantextensionacl?

**Review by fabianfett — APPROVED** [review]
LGTM. Some `internal` acls have been removed. Do we want this? Should we disable [redundantextensionacl](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#redundantextensionacl)? @tomerd wdyt?

File: Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift (hunk 3 above)
Comment (tomerd) [reply Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift:25]: meh. dialing redundantExtensionACL sounds right to me

Comment (fabianfett) [reply Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift:25]: https://github.com/swift-server/swift-aws-lambda-runtime/pull/197

**Review by fabianfett — COMMENTED** [review]
@mufumade I'm so sorry for all this `swiftformat .` confusion. Would you mind merging the current `main` branch and re-adding the dropped `internal` acls? I hope this is the last time...

**Comment by fabianfett** [comment]
@mufumade 👋 Ping: Are you still interested in getting this merged? I would love to see this land...

**Comment by mufumade** [comment]
Oh. Sorry for the delay. I'll get started right away.

**Comment by mufumade** [comment]
@fabianfett  So i merged the current main into my branch, ran swiftformat again to check but I am not quite sure what you mean by re-adding the dropped internal acls?

File: Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift (hunk 3 above)
Comment (fabianfett) [reply Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift:25]:
```suggestion
    internal struct RuntimeClient {
```

File: Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift [hunk 4]
```diff
@@ -156,7 +156,7 @@ extension ErrorResponse {
 }
 
 extension Lambda {
-    internal struct Invocation {
+    struct Invocation {
```
Comment (fabianfett) [inline Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift:159]:
```suggestion
    internal struct Invocation {
```

File: Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift [hunk 5]
```diff
@@ -194,10 +194,10 @@ extension Lambda {
 }
 
 extension Lambda.RuntimeClient {
-    internal static let defaultHeaders = HTTPHeaders([("user-agent", "Swift-Lambda/Unknown")])
+    static let defaultHeaders = HTTPHeaders([("user-agent", "Swift-Lambda/Unknown")])
```
Comment (fabianfett) [inline Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift:197]:
```suggestion
    internal static let defaultHeaders = HTTPHeaders([("user-agent", "Swift-Lambda/Unknown")])
```

File: Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift [hunk 6]
```diff
@@ -194,10 +194,10 @@ extension Lambda {
...
 
     /// These headers must be sent along an invocation or initialization error report
-    internal static let errorHeaders = HTTPHeaders([
+    static let errorHeaders = HTTPHeaders([
```
Comment (fabianfett) [inline Sources/AWSLambdaRuntimeCore/LambdaRuntimeClient.swift:200]:
```suggestion
    internal static let errorHeaders = HTTPHeaders([
```

File: Sources/AWSLambdaRuntimeCore/Utils.swift [hunk 7]
```diff
@@ -117,7 +117,7 @@ extension AmazonHeaders {
     /// # References
     /// - [Generating trace IDs](https://docs.aws.amazon.com/xray/latest/devguide/xray-api-sendingdata.html#xray-api-traceids)
     /// - [Tracing header](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader)
-    internal static func generateXRayTraceID() -> String {
+    static func generateXRayTraceID() -> String {
```
Comment (fabianfett) [inline Sources/AWSLambdaRuntimeCore/Utils.swift:120]:
```suggestion
    internal static func generateXRayTraceID() -> String {
```

File: Sources/AWSLambdaRuntimeCore/LambdaRunner.swift [hunk 8]
```diff
@@ -18,7 +18,7 @@ import NIO
 
 extension Lambda {
     /// LambdaRunner manages the Lambda runtime workflow, or business logic.
-    internal final class Runner {
+    final class Runner {
```
Comment (fabianfett) [inline Sources/AWSLambdaRuntimeCore/LambdaRunner.swift:21]:
```suggestion
    internal final class Runner {
```

File: Sources/AWSLambdaRuntimeCore/LambdaConfiguration.swift [hunk 9]
```diff
@@ -17,7 +17,7 @@ import Logging
 import NIO
 
 extension Lambda {
-    internal struct Configuration: CustomStringConvertible {
+    struct Configuration: CustomStringConvertible {
```
Comment (fabianfett) [inline Sources/AWSLambdaRuntimeCore/LambdaConfiguration.swift:20]:
```suggestion
    internal struct Configuration: CustomStringConvertible {
```

File: Sources/AWSLambdaRuntimeCore/Lambda+LocalServer.swift [hunk 10]
```diff
@@ -35,7 +35,7 @@ extension Lambda {
     ///     - body: Code to run within the context of the mock server. Typically this would be a Lambda.run function call.
     ///
     /// - note: This API is designed stricly for local testing and is behind a DEBUG flag
-    internal static func withLocalServer<Value>(invocationEndpoint: String? = nil, _ body: @escaping () -> Value) throws -> Value {
+    static func withLocalServer<Value>(invocationEndpoint: String? = nil, _ body: @escaping () -> Value) throws -> Value {
```
Comment (fabianfett) [inline Sources/AWSLambdaRuntimeCore/Lambda+LocalServer.swift:38]:
```suggestion
    internal static func withLocalServer<Value>(invocationEndpoint: String? = nil, _ body: @escaping () -> Value) throws -> Value {
```

File: Sources/AWSLambdaEvents/Utils/HTTP.swift [hunk 11]
```diff
@@ -157,7 +157,7 @@ extension HTTPResponseStatus: Codable {
 }
 
 extension String {
-    internal var isValidHTTPToken: Bool {
+    var isValidHTTPToken: Bool {
```
Comment (fabianfett) [inline Sources/AWSLambdaEvents/Utils/HTTP.swift:160]:
```suggestion
    internal var isValidHTTPToken: Bool {
```

**Review by fabianfett — COMMENTED** [review]
@mufumade I have added suggestions to re-add all the `internal` ACL that were dropped by swiftformat before. Just add those and we should be ready to go :) Thanks! And sorry for all the confusion.

**Comment by fabianfett** [comment]
@swift-server-bot test this please

**Review by tomerd — APPROVED** [review]

**Pull request merged**